flink-runtime/src/main/java/org/apache/flink/runtime/client/JobClientActor.java

```diff
@@ -167,4 +275,10 @@ private void logAndPrintMessage(Object message) {
 			System.out.println(message.toString());
 		}
 	}
+
+	private void resetContextAndActor(){
+		this.currentJobCreatedAt = -1;
+		this.currentJobId = null;
+		getContext().setReceiveTimeout(FiniteDuration.Inf());
```

Comment:
According to the docs, you should pass `Duration.Undefined` to turn off the receive timeout message.